crates/l3/appchain-core-contract-client/src/interfaces/operator.rs

```diff
@@ -0,0 +1,120 @@
+use common::LocalWalletSignerMiddleware;
+use common::{call_contract, invoke_contract};
+use starknet_accounts::Execution;
+use starknet_core::types::{FieldElement, StarknetError};
+use std::sync::Arc;
+use common::errors::Error;
+
+pub struct Operator {
+    client: Arc<LocalWalletSignerMiddleware>,
+    address: FieldElement,
+}
+
+impl Operator {
+    pub fn new(address: FieldElement, client: Arc<LocalWalletSignerMiddleware>) -> Self {
+        Self { client, address }
+    }
+
+    pub async fn register_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<Option<Execution<LocalWalletSignerMiddleware>>, Error> {
+        let execution = invoke_contract(
+            &self.client,
+            self.address,
+            "register_operator",
+            vec![new_operator.into()],
+        )
+        .await;
+
+        Ok(Some(execution))
+    }
+
+    pub async fn unregister_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<Option<Execution<LocalWalletSignerMiddleware>>, Error> {
+        let execution = invoke_contract(
+            &self.client,
+            self.address,
+            "unregister_operator",
+            vec![new_operator.into()],
+        )
+        .await;
+        Ok(Some(execution))
+    }
+
+    pub async fn is_operator(&self) -> Result<bool, Error> {
+        let result = call_contract(&self.client, self.address, "is_operator").await;
```

Comment:
we can just do await`?` and it will throw the error automatically. also do run cargo fmt and cargo clippy on the code once

Reply:
it is giving error as 
the `?` operator can only be applied to values that implement `Try`
the trait `Try` is not implemented for `Execution<'_, SingleOwnerAccount<JsonRpcClient<HttpTransport>